libs/books/feature/src/lib/reading-list/reading-list.component.ts

```diff
@@ -10,9 +17,15 @@ import { getReadingList, removeFromReadingList } from '@tmo/books/data-access';
 export class ReadingListComponent {
   readingList$ = this.store.select(getReadingList);
 
-  constructor(private readonly store: Store) {}
+  constructor(private readonly store: Store) { }
 
   removeFromReadingList(item) {
     this.store.dispatch(removeFromReadingList({ item }));
+    this.store.dispatch(init());
+  }
+
+  finishReadingList(item) {
+    this.store.dispatch(markAsFinished({ item }));
+    this.store.dispatch(init());
```

Comment:
Since state is already updated in markAsFinished action, there is no need to call the API again